### PR TITLE
[:pineapple: < T1216MG] T1216 implement aggregate

### PR DESCRIPTION
**Pull request by 42jeremy**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Provide the full content or a guide for the final git message

### CODE REVIEW

File: src/query/v2/plan/operator.cpp [hunk 1]
```diff
@@ -1233,6 +1233,55 @@ class AggregateCursor : public Cursor {
...
+
+    if (!pulled_all_input_) {
+      while (!pulled_all_input_) {
+        ProcessAll(multi_frame, &context);
+      }
+      multi_frame.MakeAllFramesInvalid();
+      aggregation_it_ = aggregation_.begin();
+
+      // in case there is no input and no group_bys we need to return true
+      // just this once
```
Comment (jbajic) [inline src/query/v2/plan/operator.cpp:1247]: This comment seems false

File: src/query/v2/plan/operator.cpp [hunk 2]
```diff
@@ -1233,6 +1233,55 @@ class AggregateCursor : public Cursor {
...
+      frame[aggregation_elem.output_sym] = *aggregation_values_it++;
+    }
+
+    // place remember values on the frame
+    auto remember_values_it = aggregation_it_->second.remember_.begin();
+    for (const Symbol &remember_sym : self_.remember_) {
+      frame[remember_sym] = *remember_values_it++;
+    }
+
+    aggregation_it_++;
```
Comment (jbajic) [inline src/query/v2/plan/operator.cpp:1282]:
```suggestion
    ++aggregation_it_;
```

File: src/query/v2/plan/operator.cpp [hunk 3]
```diff
@@ -1311,6 +1360,36 @@ class AggregateCursor : public Cursor {
...
+
+    // calculate AVG aggregations (so far they have only been summed)
+    for (size_t pos = 0; pos < self_.aggregations_.size(); ++pos) {
+      if (self_.aggregations_[pos].op != Aggregation::Op::AVG) continue;
+      for (auto &kv : aggregation_) {
+        AggregationValue &agg_value = kv.second;
+        auto count = agg_value.counts_[pos];
+        auto *pull_memory = context->evaluation_context.memory;
+        if (count > 0) {
+          agg_value.values_[pos] = agg_value.values_[pos] / TypedValue(static_cast<double>(count), pull_memory);
```
Comment (jbajic) [inline src/query/v2/plan/operator.cpp:1387]: If there are no elements we should still be able to calculate avg, but this seems as temp code, so feel free to do what you think make sense

File: src/query/v2/plan/operator.cpp (hunk 1 above)
Comment (antaljanosbenjamin) [reply src/query/v2/plan/operator.cpp:1337]: If you check the first condition (`if (!pulled_all_input_)`), that will be true only on the first run, so here we only return true only one.

File: src/query/v2/plan/operator.cpp (hunk 3 above)
Comment (antaljanosbenjamin) [reply src/query/v2/plan/operator.cpp]: After discussing it on slack we agreed that setting the aggregation value to null is fine and the default value is null (check `EnsureInitialized`), so this is fine.

**Review by jbajic — APPROVED** [review]

**Pull request merged**